### PR TITLE
Embed winston-syslog without unix-dgram

### PR DESCRIPTION
**Pull request by Stabilostick**
Avoid compiler warnings because of old nan version.

### CODE REVIEW

File: lib/setup.js [hunk 1]
```diff
@@ -2293,7 +2293,7 @@ function processCommand(command, args, params, callback) {
                 let oldHostname;
                 let newHostname;
                 if (!change) {
-                    console.warn('Please write "' + tools.appName + ' host this" to use this host ("' + tools.getHostName() + '") in ' + tools.appName + ' for all instances.');
+                    console.warn('Please write "' + tools.appName + ' host this" to use actual hostname ("' + tools.getHostName() + '") in ' + tools.appName + ' for all instances.');
```
Comment (AlCalzone) [inline lib/setup.js]: Klingt nach falschem Englisch. Wie wäre es mit "... to use the current hostname ..."?

File: lib/setup.js [hunk 2]
```diff
@@ -2344,7 +2344,7 @@ function processCommand(command, args, params, callback) {
                                 }
                             }
                             if (hostCount > 1) {
-                                console.warn('More than one host found. You must specifiy which hist must be renamed.');
+                                console.warn('More than one host found. Please write "' + tools.appName + ' host <oldHostname> to use actual hostname ("' + tools.getHostName() + '") in ' + tools.appName + ' for related instances.');
```
Comment (AlCalzone) [inline lib/setup.js]: Die Änderung find ich gut, die Formulierung nicht - verstehe ich auch nach dem zweiten Lesen nicht :sweat_smile:

Comment (Stabilostick) [reply lib/setup.js]: Habe ich wieder vom bestehenden übernommer ,-)

File: lib/setup.js (hunk 1 above)
Comment (Stabilostick) [reply lib/setup.js]: Ack.

File: lib/setup.js [hunk 3]
```diff
@@ -2373,7 +2376,7 @@ function processCommand(command, args, params, callback) {
                                 if (hName === newHostname && change !== 'remove') {
                                     hostCount++;
                                     if (hostCount > 1) {
-                                        console.error('Host with actual hostname "' + hName + '" found. Cannot rename it.');
+                                        console.error('Host with current hostname "' + hName + '" found. Cannot rename it.');
```
Comment (AlCalzone) [inline lib/setup.js]: I'm still not sure about this message

File: lib/setup.js [hunk 4]
```diff
@@ -2344,7 +2347,7 @@ function processCommand(command, args, params, callback) {
                                 }
                             }
                             if (hostCount > 1) {
-                                console.warn('More than one host found. Please write "' + tools.appName + ' host <oldHostname> to use actual hostname ("' + tools.getHostName() + '") in ' + tools.appName + ' for related instances.');
+                                console.warn('More than one host found. Specify old hostname to rename to current hostname. Use: "' + tools.appName + ' host <oldHostname>"');
```
Comment (AlCalzone) [inline lib/setup.js]: I think mentioning the target host name (`tools.getHostName()`) is fine.

**Comment by Apollon77** [comment]
Yes, a PR for the "not-windston-syslog parts" for master would be awesome. Rview done and so can be merged into 1.5?

**Comment by AlCalzone** [comment]
LGTM
Haven't tested myself though.

**Comment by Stabilostick** [comment]
Tested with Windows 10 x64 with node 10.16 and syslog server on different machine using ipv4/6 and tcp/utp. Tested „npm rebuild“ on this machine with no nan-warnings for js-controller anymore.

**Comment by Apollon77** [comment]
@GermanBluefox Changelog for this PR
* (Stabilostick) Prevent an error with status reporting
* (Stabilostick) Optimize error messages for some host CLI commands
* (Stabilostick) replace winston-sysloh dependecy to prevent build errors on nodejs 10+ when installing or updating iobroker

**Comment by Stabilostick** [comment]
Seems the error with status locking is not finally resolved :-(

**Comment by Apollon77** [comment]
Re Appveyor: To be honest: I have no idea what happens there because the code should never reach that state ... it needs to be atiming issue on appveyor/in the testing ... so ignore for now unless it happens again in 2.0.0

**Pull request merged**